arraycontext/impl/pytato/batched_einsum/utils.py

```diff
@@ -0,0 +1,647 @@
+
+            # {{{ Get the coefficients
+
+            iname1_coeff, iname2_coeff = (coeffs_by_name[iname1].to_python(),
+                                          coeffs_by_name[iname2].to_python())
+
+            # }}}
+
+            # {{{ get the tags
+
+            if transformed_kernel.iname_tags_of_type(iname1, InameImplementationTag):
+                iname1_tag, = transformed_kernel.iname_tags_of_type(
+                    iname1, InameImplementationTag)
+            else:
+                iname1_tag = None
+
+            if transformed_kernel.iname_tags_of_type(iname2, InameImplementationTag):
+                iname2_tag, = transformed_kernel.iname_tags_of_type(
+                    iname2, InameImplementationTag)
+            else:
+                iname2_tag = None
+
+            # }}}
+
+            # {{{ construct the SplitInfo
+
+            if iname1_coeff != 1 and iname2_coeff != 1:
+                raise _CannotInferSplitInfoFromFeinsum
+            elif iname1_coeff == 1:
+                split_infos.append(SplitInfo(src_iname,
+                                             iname2_coeff,
+                                             inner_tag=iname1_tag,
+                                             outer_tag=iname2_tag))
+            elif iname2_coeff == 1:
+                split_infos.append(SplitInfo(src_iname,
+                                             iname1_coeff,
+                                             inner_tag=iname2_tag,
+                                             outer_tag=iname1_tag))
+            else:
+                raise AssertionError("Control should not reach here."
+                                     " Unreachable branch.")
+
+            # }}}
+
+    return tuple(split_infos)
+
+
+def _reverse_engineer_splitting(t_unit: lp.LoopKernel,
+                                kernel_name: str,
+                                loop_nest: _LoopNest,
+                                iname_to_length: Mapping[str, Union[int, float]],
+                                cl_device: "pyopencl.Device",
+                                database: str) -> lp.LoopKernel:
+    import feinsum as fnsm
+
+    from arraycontext.impl.pytato.split_actx.utils import (
+        _get_iname_pos_from_loop_nest)
+
+    domain = t_unit[kernel_name].get_inames_domain(loop_nest.inames)
+    if frozenset(domain.get_var_dict()) != loop_nest.inames:
+        raise NotImplementedError("Current implementation requires"
+                                  " ")
+    iname_pos = _get_iname_pos_from_loop_nest(t_unit[kernel_name], loop_nest)
+    subscripts = ", ".join(sorted(loop_nest.inames, key=lambda x: iname_pos[x]))
+    subst_rule_args = ", ".join(f"_{i}" for i in range(len(loop_nest.inames)))
+
+    template_kernel = lp.make_kernel(
+        [domain],
+        f"""
+        in0({subst_rule_args}) := 3.141592653589793
+        in1({subst_rule_args}) := 2.718281828459045
```

Comment:
?

Reply:
Values for $\pi$ and $e$ upto double-precision :). No good?